test/test_linalg.py

```diff
@@ -5901,11 +5908,22 @@ def _test_addmm_addmv(self, f, t, m, v, *, alpha=None, beta=None, transpose_out=
         else:
             alpha = 1.2 if alpha is None else alpha
             beta = 0.8 if beta is None else beta
-        res1 = f(t, m, v, alpha=alpha, beta=beta)
+
+        # float16 inputs with cusparse backend require larger tolerance than with cublas
+        if layout == torch.sparse_csr and m.dtype == torch.float16:
+            self.precision = 1e-1
+
+        def to_sparse_csr(mat):
```

Comment:
This is a misnomer, because it only converts to `sparse_csr` if the respective layout is used. Maybe something like `convert_layout`?